src/libs/focusWithDelay/focusWithDelay.js

```diff
@@ -0,0 +1,40 @@
+import {InteractionManager} from 'react-native';
+
+/**
+ * Creates a function that can be used to focus a text input
+ * @param {Boolean} disableDelay whether to force focus without a delay (on web and desktop)
+ * @returns {Function} a focusWithDelay function
+ */
+function focusWithDelay(disableDelay = false) {
+    /**
+     * Create a function that focuses a text input.
+     * @param {Object} textInput the text input to focus
+     * @returns {Function} a function that focuses the text input with a configurable delay
+     */
+    return (textInput) =>
+        /**
+         * Focus the text input
+         * @param {Boolean} [shouldelay=false] Impose delay before focusing the text input
+         */
+        (shouldelay = false) => {
```

Comment:
I understand it was original name but let's rename it to `shouldDelay` here as I think it's better